flow360/component/results/case_results.py

```diff
@@ -0,0 +1,891 @@
+""" Case results module"""
+
+import os
+import re
+import shutil
+import tempfile
+import time
+import uuid
+from enum import Enum
+from typing import Callable, Dict, List, Optional
+
+import numpy as np
+import pandas
+import pydantic as pd
+
+from ...cloud.s3_utils import (
+    CloudFileNotFoundError,
+    get_local_filename_and_create_folders,
+)
+from ...exceptions import Flow360ValueError
+from ...log import log
+from ..flow360_params.conversions import unit_converter
+from ..flow360_params.flow360_params import Flow360Params
+from ..flow360_params.unit_system import (
+    Flow360UnitSystem,
+    ForceType,
+    MomentType,
+    PowerType,
+    is_flow360_unit,
+)
+
+TMP_DIR = tempfile.mkdtemp()
```

Comment:
This is not ideal because we don't clean up temp files. Need to find a better way in the future

Reply:
I suggest we move all globally-accessible temporary directories to a separate manager, call it TemporaryResourceManager and use the `__enter__` and `__exit__` methods to allocate and deallocate temporary resources as well as manage access to them for other systems, similar to how we use UserConfig / UnitSystem right now. Should we fix this now or later?

Reply:
how about error, will it free resources when errors happen?